src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -209,14 +209,16 @@ class GrowableArrayView : public GrowableArrayBase {
     return -1;
   }
 
-  int  find(void* token, bool f(void*, E)) const {
+  template<typename T, typename F>
+  int find(T* token, F f) const {
```

Comment:
Should be

    template <typename F>
    int find(F f) const {
        for (int i = 0; i < _len; i++) {
            if (f(_data[i]) {
                return i;
            }
        }
        return -1;
    }

Reply:
We need `token` to find it in the array, don't we? All the invocations pass such a function with two parameters. The change here needs all invocations to be changed.

Reply:
No, it can be embedded into the function object.

Reply:
I think you can have 2 versions, `GrowableArray::find(const E& value)` and `GrowableArray::find_if(UnaryPredicate p)`.